api/pkg/gcs/gcs.go

```diff
@@ -0,0 +1,59 @@
+package gcs
+
+import (
+	"context"
+	"strings"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+)
+
+type gcsClient struct {
+	Api    *storage.Client
+	Config Config
+}
+type Config struct {
+	Ctx context.Context
+}
+type GcsService interface {
```

Comment:
Hmm okay maybe this is due a difference in opinion in some design choice but I was originally thinking this could've just been a generic interface like `ArtifactService` and that the `gcsClient` struct could be implement this. The main motivation was so that we could initialise a new `MlflowClient` with any kind of `ArtifactService`, say an `sqlClient` or something similar. Was there a reason why you chose to call it this way?